sympy/functions/elementary/trigonometric.py

```diff
@@ -2924,7 +2924,15 @@ def _eval_rewrite_as_log(self, y, x, **kwargs):
         return -S.ImaginaryUnit*log((x + S.ImaginaryUnit*y) / sqrt(x**2 + y**2))
 
     def _eval_rewrite_as_atan(self, y, x, **kwargs):
-        return 2*atan(y / (sqrt(x**2 + y**2) + x))
+        if y.is_zero is True:
+            und = Function('undefined')
+            return Piecewise(
+                (pi, x < 0),
+                (0, x > 0),
+                (und, True)
+            )
+        else:
+            return 2 * atan(y / (sqrt(x ** 2 + y ** 2) + x))
```

Comment:
In this branch we know that y.is_zero is not True. If we knew that y.is_zero was False then the above would be correct. However we also need to consider the case that y.is_zero is None meaning we don't know whether y is zero or not. In that case we still need a Piecewise.

I wonder whether it's actually better just to construct a full piecewise and let Piecewise work out how to simplify itself with the assumptions.

Reply:
Could you please give me an example when y.is_zero is none?

Reply:
```
In [1]: y = Symbol('y')                                                                                                                        

In [2]: print(y.is_zero)                                                                                                                       
None
```
It happens when it is not known that y is zero and it is also not known that y is non-zero. Usually this shoul only be the case where symbols are involved. We can also attached assumptions to symbols so that it is known:
```
In [3]: y = Symbol('y', nonzero=True)                                                                                                          

In [4]: print(y.is_zero)                                                                                                                       
False
```